dds/DCPS/transport/multicast/MulticastDataLink.cpp

```diff
@@ -315,10 +333,89 @@ MulticastDataLink::sample_received(ReceivedDataSample& sample)
 }
 
 bool
+MulticastDataLink::ready_to_deliver(const ReceivedDataSample& data, ACE_INT32 source)
+{
+  if (data.header_.sequence_ == SequenceNumber::SEQUENCENUMBER_UNKNOWN() ||
+      data.header_.publication_id_ == GUID_UNKNOWN) {
+    return true;
+  }
+
+  ACE_GUARD_RETURN(ACE_SYNCH_RECURSIVE_MUTEX,
+      guard,
+      this->session_lock_, false);
+  SequenceNumber seq = data.header_.sequence_;
+  MulticastSessionMap::iterator session_it = sessions_.find(source);
+  if (session_it != sessions_.end()) {
+    bool reliable = session_it->second->is_reliable();
+    if (!reliable) {
+      return true;
+    }
+    MulticastSession_rch session(session_it->second);
+    DisjointSequence recvd = this->data_samples_seen_[data.header_.publication_id_];
```

Comment:
Is a copy of the DisjointSequence needed here?  That could be expensive (heap alloc).